R/chromote_session.R

```diff
@@ -10,29 +10,50 @@ ChromoteSession <- R6Class(
   lock_objects = FALSE,
   cloneable = FALSE,
   public = list(
+    #' @description Create a new `ChromoteSession` object.
+    #' @param parent [`Chromote`] object to use; defaults to
+    #'   [default_chromote_object()]
+    #' @param width Width, in pixels, of the `Target` to create if `tid` is
+    #'   `NULL`
+    #' @param height Height, in pixels, of the `Target` to create if `tid` is
+    #'   `NULL`
+    #' @param tid
+    #'   [Target](https://chromedevtools.github.io/devtools-protocol/tot/Target)
+    #'   ID of an existing target to attach to. When a `tid` is provided, the
+    #'   `width` and `height` arguments are ignored. If NULL (the default) a new
+    #'   target is created and attached to, and the `width` and `height`
+    #'   arguments determine its viewport size.
+    #' @param wait_ If `FALSE`, return a [promises::promise] of a new
+    #'   `ChromoteSession` object. Otherwise, block during initialization, and
+    #'   return a `ChromoteSession` object directly.
+    #' @return A new `ChromoteSession` object.
     initialize = function(
       parent = default_chromote_object(),
       width = 992,
       height = 744,
+      tid = NULL,
       wait_ = TRUE
     ) {
       self$parent <- parent
 
       # Create a session from the Chromote. Basically the same code as
       # new_session(), but this is synchronous.
-      p <- parent$Target$createTarget(
+      p <- (if (is.null(tid)) {
```

Comment:
Can you do this like so:

```R
if() {
  p <- ...
} else {
  p <- ...
}
```

In general, I'd prefer not to have a multi-line `if` statement inside of assignment.

Reply:
I did it. The only thing I didn't like about initializing `p` to the target was that the type of `p` then changed between assignments; I resolved it for myself by initializing a variable called `target` instead.

When I had the inline `if`, I ran across something really odd I wanted to share with you. This worked:
```R
p <- (if() {
  p <- ...
} else {
  p <- ...
})$
```
But this did not:
p <- if() {
  p <- ...
} else {
  p <- ...
}$
```
I don't know why exactly, but I think it might have something to do with precedence rules in the parser that bind the `$` to the `else` block instead of to the entire `if` expression. Anyway, 🤷‍♂